custom/abdm/tests/test_abha_creation.py

```diff
@@ -0,0 +1,85 @@
+        self.assertEqual(response.json(), abdm_txn_id_mock)
+
+    def test_aadhaar_otp_generation_failure(self):
+        response = self.client.post(reverse("generate_aadhaar_otp"), {"pan": "123456"})
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(self.invalid_req_msg, response.json().get("message"))
+
+    def test_mobile_otp_generation_success(self):
+        abdm_txn_id_mock = {"txnId": "1234"}
+        with patch('custom.abdm.milestone_one.views.abha_creation_views.abdm_util.get_response_http_post',
+                   side_effect=TestABHACreation._mock_abdm_http_post):
+            response = self.client.post(reverse("generate_mobile_otp"),
+                                        {"txn_id": "1234", "mobile_number": "9999988888"})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.json(), abdm_txn_id_mock)
+
+    def test_mobile_otp_generation_failure(self):
+        response = self.client.post(reverse("generate_mobile_otp"), {"pan": "123456"})
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(self.invalid_req_msg, response.json().get("message"))
+
+    def test_aadhaar_otp_verification_success(self):
+        abdm_txn_id_mock = {"txnId": "1234"}
+        with patch('custom.abdm.milestone_one.views.abha_creation_views.abdm_util.get_response_http_post',
+                   side_effect=TestABHACreation._mock_abdm_http_post):
+            response = self.client.post(reverse("verify_aadhaar_otp"),
+                                        {"txn_id": "1234", "otp": "1111"})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.json(), abdm_txn_id_mock)
+
+    def test_aadhaar_otp_verification_failure(self):
+        response = self.client.post(reverse("verify_aadhaar_otp"), {"pan": "123456"})
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(self.invalid_req_msg, response.json().get("message"))
+
+    def test_mobile_otp_verification_success(self):
+        with patch('custom.abdm.milestone_one.views.abha_creation_views.abdm_util.get_response_http_post',
+                   side_effect=TestABHACreation._mock_abdm_http_post):
+            response = self.client.post(reverse("verify_mobile_otp"),
+                                        {"txn_id": "1234", "otp": "1111"})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.json(), {"health_id": "123-456", "txnId": "1234"})
+
+    def test_mobile_otp_verification_failure(self):
+        response = self.client.post(reverse("verify_mobile_otp"), {"pan": "123456"})
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(self.invalid_req_msg, response.json().get("message"))
+
+    @staticmethod
+    def _mock_abdm_http_post(url, payload):
```

Comment:
nit: just a coding practice that I find nice, is to have such methods at the top before all the tests begin.
This also makes it easier to review.

Reply:
https://github.com/dimagi/commcare-hq/pull/32332/commits/0b755dd44470292cbeda6c6ed7e11657449b86b5